ExpDev07/coronavirus-tracker-api
# Add testsuite for unit- and integration tests

Based on #64, I created a basic test suite with following features:
* Unit-Testing of all help function, like `date.is_date`
* Integration-Testing of API endpoints (validation agains output from `tests/expected_output`)
* Add example data to `tests/example_data` for testing
* Mocking of `app.data.requests.get` and `app.data.datetime`  to enforce data fetch locally from `tests/example_data`
* Edit of `README.md` for testing and linting
* Add `pytest` and `pytest` to `Pipfile`  to `dev-packages`
* Randomize testing possible to test for stable code
* Update `Pipfile.lock` with new dev-dependencies

Testing of PR:
``` bash
$ pytest -s -v tests/
```

ToDos:
* Add additonal testcases for endpoint `/all`. It is hard to test due to caching mechanism. Need to check, if it is possible to deactivate caching during testing
* Optional: Contol testing per `Makefile`
* Optional: Add to CI-pipeline
